src/devices/Mlx90640/Mlx90640.cs

```diff
@@ -0,0 +1,957 @@
+        private float _cpKta = 0;
+        private float _cpKv = 0;
+        private int[] _cpOffset = new int[2];
+        private int[] _ct = new int[5];
+        private short _gainEE = 0;
+        private float[] _ilChessC = new float[3];
+        private float _ksTa = 0;
+        private double[] _ksTo = new double[5];
+        private byte[] _kta = new byte[PixelCount];
+        private byte _ktaScale = 0;
+        private float _ktPTAT = 0;
+        private short _kVdd = 0;
+        private sbyte[] _kv = new sbyte[PixelCount];
+        private float _kvPTAT = 0;
+        private byte _kvScale = 0;
+        private int[] _offset = new int[PixelCount];
+        private ushort[] _outlierPixels = new ushort[5];
+        private byte _resolutionEE = 0;
+        private float _tgc = 0;
+        private short _vdd25 = 0;
+        private short _vPTAT25 = 0;
+
+        /// <summary>
+        /// MLX90640 Default I2C Address
+        /// </summary>
+        public const byte DefaultI2cAddress = 0x33;
+
+        /// <summary>
+        /// Creates a new instance of the MLX90640
+        /// </summary>
+        /// <param name="i2cDevice">The I2C device used for communication.</param>
+        public Mlx90640(I2cDevice i2cDevice)
+        {
+            _i2cDevice = i2cDevice ?? throw new ArgumentNullException(nameof(i2cDevice));
+            DumpEE();
+            ExtractVDDParameters();
+            ExtractPTATParameters();
+            ExtractVDDParameters();
+            ExtractPTATParameters();
+            ExtractGainParameters();
+            ExtractTgcParameters();
+            ExtractResolutionParameters();
+            ExtractKsTaParameters();
+            ExtractKsToParameters();
+            ExtractCPParameters();
+            ExtractAlphaParameters();
+            ExtractOffsetParameters();
+            ExtractKtaPixelParameters();
+            ExtractKvPixelParameters();
+            ExtractCILCParameters();
+            ExtractDeviatingPixels();
+        }
+
+        /// <summary>
+        /// Sets sampling to the given value
+        /// </summary>
+        /// <param name="mode">Sampling Mode</param>
+        public void SetSampling(Sampling mode)
+        {
+            Span<ushort> controlRegister1 = stackalloc ushort[1];
+
+            int value = ((byte)mode & 0x07) << 7;
+
+            controlRegister1 = ReadRegisters16(0x800D, 1);
+            value = (controlRegister1[0] & 0xFC7F) | value;
+            Write16(0x800D, value);
+        }
+
+        /// <summary>
+        /// Request both 'halves' of a frame from the sensor, merge them
+        /// and calculate the temperature in C for each of 32x24 pixels.
+        /// </summary>
+        /// <returns>Temperature of the specified pixel</returns>
+        public Temperature[] GetFrame()
+        {
+            Temperature[] framebuf = new Temperature[PixelCount];
+            float emissivity = 0.95f;
+            ushort[] mlx90640Frame = new ushort[834];
```

Comment:
Why 834?

Reply:
That's is the allocation memory recommended in the datasheet.